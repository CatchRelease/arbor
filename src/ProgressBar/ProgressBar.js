/** @jsx jsx */
import { css, jsx } from '@emotion/core';
import styled from '@emotion/styled';
import PropTypes from 'prop-types';

import Box from '../Box';
import Flex from '../Flex';
import Icon from '../Icon';
import Text from '../Text';

const ProgressBarContainer = styled(Flex)`
  border-radius: ${props => props.theme.radii.large};
  height: 16px;
  max-width: 100%;
  align-items: center;
`;

const BarBackground = styled(Box)`
  flex: 1;
  background-color: ${props => props.theme.colors.monochrome.grey40};
  height: ${props => props.theme.space.smallest};
  border-radius: inherit;
`;
const BarForeground = styled(Box)`
  height: 100%;
  background-color: ${props => props.theme.colors.text.default};
  border-radius: inherit;
`;

const percentageText = percentage => (
  <Text flex="0" ml="smaller" fontSize="size3" fontWeight="medium">
    {`${percentage}%`}
  </Text>
);

const ProgressBar = ({ percentage, iconEndName, showPercentage, ...props }) => (
  <ProgressBarContainer {...props}>
    <BarBackground className="bar-background">
      <BarForeground
        className="bar-foreground"
        css={css`
          width: ${percentage}%;
        `}
      />
    </BarBackground>
    {iconEndName && (
      <Icon
        ml="smaller"
        flex="0"
        fontSize="size4"
        name={iconEndName}
        color={percentage < 100 ? 'monochrome.grey40' : 'text.default'}
      />
    )}
    {showPercentage && percentageText(percentage)}
  </ProgressBarContainer>
);

ProgressBar.propTypes = {
  /* Show the percentage complete, or not */
  showPercentage: PropTypes.bool,

  /* Name of the icon to use as the indicator, if desired. */
  iconEndName: PropTypes.string,

  /* The actual current percentage completed */
  percentage: PropTypes.number.isRequired
};

ProgressBar.defaultProps = {
  ...Flex.defaultProps,
  showPercentage: false,
  iconEndName: null
};

export default ProgressBar;
