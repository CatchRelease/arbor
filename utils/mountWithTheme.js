import { mount } from 'enzyme';
import { ThemeProvider } from '@emotion/react';
import theme from '../src/theme';

export default (Component, mountOptions = {}) => {
  const wrappedComponent = (
    <ThemeProvider theme={theme}>{Component}</ThemeProvider>
  );

  const mounted = mount(wrappedComponent, mountOptions);

  return mounted.find(Component.type);
};
