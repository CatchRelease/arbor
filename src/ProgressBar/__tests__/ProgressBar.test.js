import React from 'react';
import mountWithTheme from '../../../utils/mountWithTheme';

import ProgressBar from '../ProgressBar';

describe('<ProgressBar />', () => {
  it('renders the progress bar properly', () => {
    const progressBar = mountWithTheme(
      <ProgressBar
        percentage={42}
        iconEndName="calendar"
        showPercentage={false}
      />
    );

    expect(progressBar).toMatchSnapshot();
  });

  it('renders the progress bar properly at 100% with an icon', () => {
    const progressBar = mountWithTheme(
      <ProgressBar
        percentage={100}
        iconEndName="calendar"
        showPercentage={false}
      />
    );

    expect(progressBar).toMatchSnapshot();
  });
});
