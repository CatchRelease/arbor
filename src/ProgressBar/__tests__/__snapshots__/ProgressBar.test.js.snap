// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProgressBar /> renders the progress bar properly 1`] = `
.emotion-7 {
  font-family: Motiva Sans;
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  border-radius: 6px;
  height: 16px;
  max-width: 100%;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-3 {
  font-family: Motiva Sans;
  box-sizing: border-box;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  background-color: #E0E1E2;
  height: 0.25rem;
  border-radius: inherit;
}

.emotion-2 {
  width: 42%;
}

.emotion-0 {
  font-family: Motiva Sans;
  box-sizing: border-box;
  height: 100%;
  background-color: #25272D;
  border-radius: inherit;
  width: 42%;
}

.emotion-5 {
  color: #E0E1E2;
  font-size: 0.875rem;
  margin-left: 0.5rem;
}

<ProgressBar
  iconEndName="calendar"
  percentage={42}
  showPercentage={false}
>
  <ProgressBarContainer>
    <div
      className="emotion-7 emotion-8"
    >
      <BarBackground
        className="bar-background"
      >
        <div
          className="bar-background emotion-3 emotion-4"
        >
          <EmotionCssPropInternal
            __EMOTION_LABEL_PLEASE_DO_NOT_USE__="ProgressBar"
            __EMOTION_TYPE_PLEASE_DO_NOT_USE__={
              Object {
                "$$typeof": Symbol(react.forward_ref),
                "__emotion_base": "div",
                "__emotion_forwardProp": undefined,
                "__emotion_real": [Circular],
                "__emotion_styles": Array [
                  "label:Box;",
                  "label:Text;",
                  [Function],
                  ";",
                  [Function],
                  ";",
                  [Function],
                  ";",
                  [Function],
                  ";",
                  [Function],
                  ";",
                  [Function],
                  ";",
                  [Function],
                  ";",
                  [Function],
                  ";",
                  [Function],
                  ";",
                  [Function],
                  ";",
                  [Function],
                  ";",
                  [Function],
                  ";",
                  [Function],
                  ";",
                  "label:Box;",
                  "box-sizing:border-box;",
                  [Function],
                  ";",
                  [Function],
                  ";",
                  [Function],
                  ";",
                  [Function],
                  ";",
                  [Function],
                  ";",
                  [Function],
                  ";",
                  [Function],
                  ";",
                  [Function],
                  ";",
                  [Function],
                  ";",
                  "label:BarForeground;",
                  "height:100%;background-color:",
                  [Function],
                  ";border-radius:inherit;",
                ],
                "defaultProps": undefined,
                "render": [Function],
                "withComponent": [Function],
              }
            }
            className="bar-foreground"
            css={
              Object {
                "map": undefined,
                "name": "1rvwrzh-ProgressBar",
                "next": undefined,
                "styles": "width:42%;;label:ProgressBar;",
                "toString": [Function],
              }
            }
          >
            <BarForeground
              className="bar-foreground emotion-2"
            >
              <div
                className="bar-foreground emotion-0 emotion-1"
              />
            </BarForeground>
          </EmotionCssPropInternal>
        </div>
      </BarBackground>
      <ForwardRef
        color="monochrome.grey40"
        flex="0"
        fontSize="size4"
        ml="smaller"
        name="calendar"
      >
        <WithTheme(Icon)
          color="monochrome.grey40"
          flex="0"
          fontSize="size4"
          iconRef={null}
          ml="smaller"
          name="calendar"
        >
          <Icon
            color="monochrome.grey40"
            flex="0"
            fontSize="size4"
            iconRef={null}
            ml="smaller"
            name="calendar"
            theme={
              Object {
                "avatarFontSizes": Object {
                  "default": "0.6875rem",
                  "medium": "0.75rem",
                },
                "avatarSizes": Object {
                  "default": "24px",
                  "medium": "32px",
                },
                "borderWidth": Object {
                  "large": "4px",
                  "medium": "2px",
                  "small": "1px",
                },
                "borders": Object {
                  "default": "1px solid #E0E1E2",
                },
                "brandFont": "Motiva Sans",
                "breakpoints": Array [
                  "512px",
                  "768px",
                  "1024px",
                  "1600px",
                ],
                "breakpointsMap": Object {
                  "lg": "1024px",
                  "md": "768px",
                  "sm": "512px",
                  "xl": "1600px",
                },
                "buttons": Object {
                  "sizes": Object {
                    "jumbo": Object {
                      "fontSize": "0.875rem",
                      "height": "3rem",
                      "i": Object {
                        "&:first-child": Object {
                          "ml": "-1.5rem",
                        },
                        "&:last-child": Object {
                          "mr": "-1.5rem",
                        },
                        "fontSize": "1rem",
                        "width": "3rem",
                      },
                      "p": "0 1.5rem",
                    },
                    "large": Object {
                      "fontSize": "0.875rem",
                      "height": "2.5rem",
                      "i": Object {
                        "&:first-child": Object {
                          "ml": "-1.25rem",
                        },
                        "&:last-child": Object {
                          "mr": "-1.25rem",
                        },
                        "fontSize": "0.875rem",
                        "width": "2.5rem",
                      },
                      "p": "0 1.25rem",
                    },
                    "medium": Object {
                      "fontSize": "0.8125rem",
                      "height": "2rem",
                      "i": Object {
                        "&:first-child": Object {
                          "ml": "-1rem",
                        },
                        "&:last-child": Object {
                          "mr": "-1rem",
                        },
                        "fontSize": "0.75rem",
                        "width": "2rem",
                      },
                      "p": "0 1rem",
                    },
                    "small": Object {
                      "fontSize": "0.75rem",
                      "height": "1.5rem",
                      "i": Object {
                        "&:first-child": Object {
                          "ml": "-0.75rem",
                        },
                        "&:last-child": Object {
                          "mr": "-0.75rem",
                        },
                        "fontSize": "0.6875rem",
                        "width": "1.5rem",
                      },
                      "p": "0 0.75rem",
                    },
                  },
                },
                "colors": Object {
                  "background": Object {
                    "default": "#F2F2F2",
                    "disabled": "#EBECEC",
                    "muted": "#F8F9F9",
                    "white": "#FFFFFF",
                  },
                  "black": "#000000",
                  "blue": "#4466EE",
                  "blueDark": "#4455CC",
                  "blueDarkest": "#2E374E",
                  "blueLight": "#5577FF",
                  "blueLighter": "#6688FF",
                  "border": Object {
                    "dark": "#5B5D61",
                    "default": "#E0E1E2",
                    "muted": "#EBECEC",
                  },
                  "brand": Object {
                    "dark": "#081D17",
                    "darker": "#081D17",
                    "default": "#081D17",
                    "light": "#B6EFD0",
                    "lighter": "#C0F9DA",
                  },
                  "bronze": "#CC7700",
                  "bronzeDark": "#AA5500",
                  "bronzeLight": "#CC9900",
                  "green": "#00AA44",
                  "greenDark": "#008833",
                  "greenLight": "#00BB44",
                  "greenLighter": "#00CC66",
                  "grey10": "#F8F9F9",
                  "grey100": "#0F0F10",
                  "grey20": "#F2F2F2",
                  "grey30": "#EBECEC",
                  "grey40": "#E0E1E2",
                  "grey50": "#BEBEC0",
                  "grey60": "#989A9C",
                  "grey70": "#78797D",
                  "grey80": "#5B5D61",
                  "grey90": "#25272D",
                  "icon": Object {
                    "default": "#989A9C",
                    "disabled": "rgba(190,190,192,0.7)",
                  },
                  "intent": Object {
                    "brand": Object {
                      "dark": "#081D17",
                      "darker": "#081D17",
                      "default": "#081D17",
                      "light": "#B6EFD0",
                      "lighter": "#C0F9DA",
                    },
                    "danger": Object {
                      "dark": "#6c1c1c",
                      "darker": "#2b0b0b",
                      "default": "#D93838",
                      "light": "#f3c3c3",
                      "lighter": "#fbebeb",
                    },
                    "info": Object {
                      "dark": "#213175",
                      "darker": "#0d132e",
                      "default": "#4263EB",
                      "light": "#c6d0f9",
                      "lighter": "#eceffd",
                    },
                    "success": Object {
                      "dark": "#14431d",
                      "darker": "#081a0b",
                      "default": "#29863A",
                      "light": "#bedac3",
                      "lighter": "#e9f2eb",
                    },
                    "warning": Object {
                      "dark": "#622902",
                      "darker": "#271000",
                      "default": "#C55205",
                      "light": "#edcbb4",
                      "lighter": "#f9ede6",
                    },
                  },
                  "monochrome": Object {
                    "black": "#000000",
                    "grey10": "#F8F9F9",
                    "grey100": "#0F0F10",
                    "grey20": "#F2F2F2",
                    "grey30": "#EBECEC",
                    "grey40": "#E0E1E2",
                    "grey50": "#BEBEC0",
                    "grey60": "#989A9C",
                    "grey70": "#78797D",
                    "grey80": "#5B5D61",
                    "grey90": "#25272D",
                    "white": "#FFFFFF",
                  },
                  "palette": Object {
                    "blue": Object {
                      "dark": "#213175",
                      "darker": "#0d132e",
                      "default": "#4263EB",
                      "light": "#c6d0f9",
                      "lighter": "#eceffd",
                    },
                    "brand": Object {
                      "dark": "#040e0b",
                      "darker": "#010504",
                      "default": "#081D17",
                      "light": "#b4bbb9",
                      "lighter": "#e6e8e7",
                    },
                    "cyan": Object {
                      "dark": "#063f48",
                      "darker": "#02191c",
                      "default": "#0D7F91",
                      "light": "#b6d8de",
                      "lighter": "#e6f2f4",
                    },
                    "green": Object {
                      "dark": "#14431d",
                      "darker": "#081a0b",
                      "default": "#29863A",
                      "light": "#bedac3",
                      "lighter": "#e9f2eb",
                    },
                    "neutral": Object {
                      "dark": "#070708",
                      "darker": "#020203",
                      "default": "#0F0F10",
                      "light": "#b7b7b7",
                      "lighter": "#e7e7e7",
                    },
                    "orange": Object {
                      "dark": "#622902",
                      "darker": "#271000",
                      "default": "#C55205",
                      "light": "#edcbb4",
                      "lighter": "#f9ede6",
                    },
                    "pink": Object {
                      "dark": "#6b1936",
                      "darker": "#2a0a15",
                      "default": "#D6336C",
                      "light": "#f2c1d2",
                      "lighter": "#faeaf0",
                    },
                    "purple": Object {
                      "dark": "#382474",
                      "darker": "#160e2e",
                      "default": "#7048E8",
                      "light": "#d4c8f8",
                      "lighter": "#f0ecfc",
                    },
                    "red": Object {
                      "dark": "#6c1c1c",
                      "darker": "#2b0b0b",
                      "default": "#D93838",
                      "light": "#f3c3c3",
                      "lighter": "#fbebeb",
                    },
                    "teal": Object {
                      "dark": "#044231",
                      "darker": "#011a13",
                      "default": "#098562",
                      "light": "#b5dacf",
                      "lighter": "#e6f2ef",
                    },
                    "yellow": Object {
                      "dark": "#513400",
                      "darker": "#201400",
                      "default": "#A36900",
                      "light": "#e3d2b2",
                      "lighter": "#f5f0e5",
                    },
                  },
                  "primary": Object {
                    "blue": "#4263EB",
                    "cyan": "#0D7F91",
                    "green": "#29863A",
                    "neutral": "#0F0F10",
                    "orange": "#C55205",
                    "pink": "#D6336C",
                    "purple": "#7048E8",
                    "red": "#D93838",
                    "teal": "#098562",
                    "yellow": "#A36900",
                  },
                  "red": "#EE2200",
                  "redDark": "#DD1100",
                  "redLight": "#FF4444",
                  "redLighter": "#FF6655",
                  "secondary": Object {
                    "blue": "#4B9BFF",
                    "cyan": "#5FC8FF",
                    "green": "#9BE169",
                    "orange": "#FFB46E",
                    "pink": "#FF73C3",
                    "purple": "#AF78FF",
                    "red": "#FF5F5F",
                    "teal": "#50D2A0",
                    "yellow": "#FFD76E",
                  },
                  "shadow": Object {
                    "blurry": "rgba(37,39,45,0.3)",
                    "border": "rgba(37,39,45,0.47)",
                  },
                  "text": Object {
                    "dark": "#0F0F10",
                    "default": "#25272D",
                    "disabled": "rgba(120,121,125,0.7)",
                    "muted": "#5B5D61",
                    "placeholder": "#989A9C",
                  },
                  "white": "#FFFFFF",
                },
                "fontSizes": Object {
                  "input": Object {
                    "default": "0.875rem",
                    "mobile": "1rem",
                  },
                  "size1": "0.6875rem",
                  "size2": "0.75rem",
                  "size3": "0.8125rem",
                  "size4": "0.875rem",
                  "size5": "1rem",
                  "size6": "1.25rem",
                  "size7": "1.5rem",
                  "size8": "1.75rem",
                  "size9": "2.125rem",
                },
                "fontWeights": Object {
                  "bold": 700,
                  "medium": 500,
                  "regular": 400,
                },
                "fonts": Object {
                  "default": "Motiva Sans",
                  "icon": "Arbor",
                },
                "iconFontPrefix": "ar",
                "icons": Object {
                  "annotation": "039",
                  "calendar": "035",
                  "caret": "037",
                  "chatBubbleOutline": "003",
                  "checkboxChecked": "030",
                  "checkboxIndeterminate": "02f",
                  "checkboxMarkedCircleOutline": "000",
                  "chevron": "03a",
                  "crLogo": "02e",
                  "cross": "03b",
                  "download": "004",
                  "swap": "038",
                  "symbolCircle": "034",
                  "thumbDownOutline": "001",
                  "thumbUpOutline": "002",
                  "trash": "036",
                },
                "lineHeights": Object {
                  "large": 1.5,
                  "regular": 1,
                  "small": 1.25,
                },
                "radii": Object {
                  "large": "6px",
                  "larger": "12px",
                  "small": "3px",
                },
                "shadows": Object {
                  "elevation0": "0 0 1px rgba(37,39,45,0.47)",
                  "elevation1": "0 0 1px rgba(37,39,45,0.47), 0 2px 4px -2px rgba(37,39,45,0.3)",
                  "elevation2": "0 0 1px rgba(37,39,45,0.47), 0 5px 8px -4px rgba(37,39,45,0.3)",
                  "elevation3": "0 0 1px rgba(37,39,45,0.47), 0 8px 10px -4px rgba(37,39,45,0.3)",
                  "elevation4": "0 0 1px rgba(37,39,45,0.47), 0 16px 24px -8px rgba(37,39,45,0.3)",
                },
                "space": Object {
                  "large": "1.5rem",
                  "larger": "2rem",
                  "largest": "4rem",
                  "regular": "1rem",
                  "small": "0.75rem",
                  "smaller": "0.5rem",
                  "smallest": "0.25rem",
                },
              }
            }
          >
            <EmotionClassNames>
              <StyledIcon
                className="ar ar-calendar"
                color="monochrome.grey40"
                flex="0"
                fontSize="size4"
                ml="smaller"
                rotation={null}
              >
                <i
                  className="ar ar-calendar emotion-5 emotion-6"
                  color="monochrome.grey40"
                  fontSize="size4"
                />
              </StyledIcon>
            </EmotionClassNames>
          </Icon>
        </WithTheme(Icon)>
      </ForwardRef>
    </div>
  </ProgressBarContainer>
</ProgressBar>
`;

exports[`<ProgressBar /> renders the progress bar properly at 100% with an icon 1`] = `
.emotion-7 {
  font-family: Motiva Sans;
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  border-radius: 6px;
  height: 16px;
  max-width: 100%;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-3 {
  font-family: Motiva Sans;
  box-sizing: border-box;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  background-color: #E0E1E2;
  height: 0.25rem;
  border-radius: inherit;
}

.emotion-2 {
  width: 100%;
}

.emotion-0 {
  font-family: Motiva Sans;
  box-sizing: border-box;
  height: 100%;
  background-color: #25272D;
  border-radius: inherit;
  width: 100%;
}

.emotion-5 {
  color: #25272D;
  font-size: 0.875rem;
  margin-left: 0.5rem;
}

<ProgressBar
  iconEndName="calendar"
  percentage={100}
  showPercentage={false}
>
  <ProgressBarContainer>
    <div
      className="emotion-7 emotion-8"
    >
      <BarBackground
        className="bar-background"
      >
        <div
          className="bar-background emotion-3 emotion-4"
        >
          <EmotionCssPropInternal
            __EMOTION_LABEL_PLEASE_DO_NOT_USE__="ProgressBar"
            __EMOTION_TYPE_PLEASE_DO_NOT_USE__={
              Object {
                "$$typeof": Symbol(react.forward_ref),
                "__emotion_base": "div",
                "__emotion_forwardProp": undefined,
                "__emotion_real": [Circular],
                "__emotion_styles": Array [
                  "label:Box;",
                  "label:Text;",
                  [Function],
                  ";",
                  [Function],
                  ";",
                  [Function],
                  ";",
                  [Function],
                  ";",
                  [Function],
                  ";",
                  [Function],
                  ";",
                  [Function],
                  ";",
                  [Function],
                  ";",
                  [Function],
                  ";",
                  [Function],
                  ";",
                  [Function],
                  ";",
                  [Function],
                  ";",
                  [Function],
                  ";",
                  "label:Box;",
                  "box-sizing:border-box;",
                  [Function],
                  ";",
                  [Function],
                  ";",
                  [Function],
                  ";",
                  [Function],
                  ";",
                  [Function],
                  ";",
                  [Function],
                  ";",
                  [Function],
                  ";",
                  [Function],
                  ";",
                  [Function],
                  ";",
                  "label:BarForeground;",
                  "height:100%;background-color:",
                  [Function],
                  ";border-radius:inherit;",
                ],
                "defaultProps": undefined,
                "render": [Function],
                "withComponent": [Function],
              }
            }
            className="bar-foreground"
            css={
              Object {
                "map": undefined,
                "name": "1v7bxob-ProgressBar",
                "next": undefined,
                "styles": "width:100%;;label:ProgressBar;",
                "toString": [Function],
              }
            }
          >
            <BarForeground
              className="bar-foreground emotion-2"
            >
              <div
                className="bar-foreground emotion-0 emotion-1"
              />
            </BarForeground>
          </EmotionCssPropInternal>
        </div>
      </BarBackground>
      <ForwardRef
        color="text.default"
        flex="0"
        fontSize="size4"
        ml="smaller"
        name="calendar"
      >
        <WithTheme(Icon)
          color="text.default"
          flex="0"
          fontSize="size4"
          iconRef={null}
          ml="smaller"
          name="calendar"
        >
          <Icon
            color="text.default"
            flex="0"
            fontSize="size4"
            iconRef={null}
            ml="smaller"
            name="calendar"
            theme={
              Object {
                "avatarFontSizes": Object {
                  "default": "0.6875rem",
                  "medium": "0.75rem",
                },
                "avatarSizes": Object {
                  "default": "24px",
                  "medium": "32px",
                },
                "borderWidth": Object {
                  "large": "4px",
                  "medium": "2px",
                  "small": "1px",
                },
                "borders": Object {
                  "default": "1px solid #E0E1E2",
                },
                "brandFont": "Motiva Sans",
                "breakpoints": Array [
                  "512px",
                  "768px",
                  "1024px",
                  "1600px",
                ],
                "breakpointsMap": Object {
                  "lg": "1024px",
                  "md": "768px",
                  "sm": "512px",
                  "xl": "1600px",
                },
                "buttons": Object {
                  "sizes": Object {
                    "jumbo": Object {
                      "fontSize": "0.875rem",
                      "height": "3rem",
                      "i": Object {
                        "&:first-child": Object {
                          "ml": "-1.5rem",
                        },
                        "&:last-child": Object {
                          "mr": "-1.5rem",
                        },
                        "fontSize": "1rem",
                        "width": "3rem",
                      },
                      "p": "0 1.5rem",
                    },
                    "large": Object {
                      "fontSize": "0.875rem",
                      "height": "2.5rem",
                      "i": Object {
                        "&:first-child": Object {
                          "ml": "-1.25rem",
                        },
                        "&:last-child": Object {
                          "mr": "-1.25rem",
                        },
                        "fontSize": "0.875rem",
                        "width": "2.5rem",
                      },
                      "p": "0 1.25rem",
                    },
                    "medium": Object {
                      "fontSize": "0.8125rem",
                      "height": "2rem",
                      "i": Object {
                        "&:first-child": Object {
                          "ml": "-1rem",
                        },
                        "&:last-child": Object {
                          "mr": "-1rem",
                        },
                        "fontSize": "0.75rem",
                        "width": "2rem",
                      },
                      "p": "0 1rem",
                    },
                    "small": Object {
                      "fontSize": "0.75rem",
                      "height": "1.5rem",
                      "i": Object {
                        "&:first-child": Object {
                          "ml": "-0.75rem",
                        },
                        "&:last-child": Object {
                          "mr": "-0.75rem",
                        },
                        "fontSize": "0.6875rem",
                        "width": "1.5rem",
                      },
                      "p": "0 0.75rem",
                    },
                  },
                },
                "colors": Object {
                  "background": Object {
                    "default": "#F2F2F2",
                    "disabled": "#EBECEC",
                    "muted": "#F8F9F9",
                    "white": "#FFFFFF",
                  },
                  "black": "#000000",
                  "blue": "#4466EE",
                  "blueDark": "#4455CC",
                  "blueDarkest": "#2E374E",
                  "blueLight": "#5577FF",
                  "blueLighter": "#6688FF",
                  "border": Object {
                    "dark": "#5B5D61",
                    "default": "#E0E1E2",
                    "muted": "#EBECEC",
                  },
                  "brand": Object {
                    "dark": "#081D17",
                    "darker": "#081D17",
                    "default": "#081D17",
                    "light": "#B6EFD0",
                    "lighter": "#C0F9DA",
                  },
                  "bronze": "#CC7700",
                  "bronzeDark": "#AA5500",
                  "bronzeLight": "#CC9900",
                  "green": "#00AA44",
                  "greenDark": "#008833",
                  "greenLight": "#00BB44",
                  "greenLighter": "#00CC66",
                  "grey10": "#F8F9F9",
                  "grey100": "#0F0F10",
                  "grey20": "#F2F2F2",
                  "grey30": "#EBECEC",
                  "grey40": "#E0E1E2",
                  "grey50": "#BEBEC0",
                  "grey60": "#989A9C",
                  "grey70": "#78797D",
                  "grey80": "#5B5D61",
                  "grey90": "#25272D",
                  "icon": Object {
                    "default": "#989A9C",
                    "disabled": "rgba(190,190,192,0.7)",
                  },
                  "intent": Object {
                    "brand": Object {
                      "dark": "#081D17",
                      "darker": "#081D17",
                      "default": "#081D17",
                      "light": "#B6EFD0",
                      "lighter": "#C0F9DA",
                    },
                    "danger": Object {
                      "dark": "#6c1c1c",
                      "darker": "#2b0b0b",
                      "default": "#D93838",
                      "light": "#f3c3c3",
                      "lighter": "#fbebeb",
                    },
                    "info": Object {
                      "dark": "#213175",
                      "darker": "#0d132e",
                      "default": "#4263EB",
                      "light": "#c6d0f9",
                      "lighter": "#eceffd",
                    },
                    "success": Object {
                      "dark": "#14431d",
                      "darker": "#081a0b",
                      "default": "#29863A",
                      "light": "#bedac3",
                      "lighter": "#e9f2eb",
                    },
                    "warning": Object {
                      "dark": "#622902",
                      "darker": "#271000",
                      "default": "#C55205",
                      "light": "#edcbb4",
                      "lighter": "#f9ede6",
                    },
                  },
                  "monochrome": Object {
                    "black": "#000000",
                    "grey10": "#F8F9F9",
                    "grey100": "#0F0F10",
                    "grey20": "#F2F2F2",
                    "grey30": "#EBECEC",
                    "grey40": "#E0E1E2",
                    "grey50": "#BEBEC0",
                    "grey60": "#989A9C",
                    "grey70": "#78797D",
                    "grey80": "#5B5D61",
                    "grey90": "#25272D",
                    "white": "#FFFFFF",
                  },
                  "palette": Object {
                    "blue": Object {
                      "dark": "#213175",
                      "darker": "#0d132e",
                      "default": "#4263EB",
                      "light": "#c6d0f9",
                      "lighter": "#eceffd",
                    },
                    "brand": Object {
                      "dark": "#040e0b",
                      "darker": "#010504",
                      "default": "#081D17",
                      "light": "#b4bbb9",
                      "lighter": "#e6e8e7",
                    },
                    "cyan": Object {
                      "dark": "#063f48",
                      "darker": "#02191c",
                      "default": "#0D7F91",
                      "light": "#b6d8de",
                      "lighter": "#e6f2f4",
                    },
                    "green": Object {
                      "dark": "#14431d",
                      "darker": "#081a0b",
                      "default": "#29863A",
                      "light": "#bedac3",
                      "lighter": "#e9f2eb",
                    },
                    "neutral": Object {
                      "dark": "#070708",
                      "darker": "#020203",
                      "default": "#0F0F10",
                      "light": "#b7b7b7",
                      "lighter": "#e7e7e7",
                    },
                    "orange": Object {
                      "dark": "#622902",
                      "darker": "#271000",
                      "default": "#C55205",
                      "light": "#edcbb4",
                      "lighter": "#f9ede6",
                    },
                    "pink": Object {
                      "dark": "#6b1936",
                      "darker": "#2a0a15",
                      "default": "#D6336C",
                      "light": "#f2c1d2",
                      "lighter": "#faeaf0",
                    },
                    "purple": Object {
                      "dark": "#382474",
                      "darker": "#160e2e",
                      "default": "#7048E8",
                      "light": "#d4c8f8",
                      "lighter": "#f0ecfc",
                    },
                    "red": Object {
                      "dark": "#6c1c1c",
                      "darker": "#2b0b0b",
                      "default": "#D93838",
                      "light": "#f3c3c3",
                      "lighter": "#fbebeb",
                    },
                    "teal": Object {
                      "dark": "#044231",
                      "darker": "#011a13",
                      "default": "#098562",
                      "light": "#b5dacf",
                      "lighter": "#e6f2ef",
                    },
                    "yellow": Object {
                      "dark": "#513400",
                      "darker": "#201400",
                      "default": "#A36900",
                      "light": "#e3d2b2",
                      "lighter": "#f5f0e5",
                    },
                  },
                  "primary": Object {
                    "blue": "#4263EB",
                    "cyan": "#0D7F91",
                    "green": "#29863A",
                    "neutral": "#0F0F10",
                    "orange": "#C55205",
                    "pink": "#D6336C",
                    "purple": "#7048E8",
                    "red": "#D93838",
                    "teal": "#098562",
                    "yellow": "#A36900",
                  },
                  "red": "#EE2200",
                  "redDark": "#DD1100",
                  "redLight": "#FF4444",
                  "redLighter": "#FF6655",
                  "secondary": Object {
                    "blue": "#4B9BFF",
                    "cyan": "#5FC8FF",
                    "green": "#9BE169",
                    "orange": "#FFB46E",
                    "pink": "#FF73C3",
                    "purple": "#AF78FF",
                    "red": "#FF5F5F",
                    "teal": "#50D2A0",
                    "yellow": "#FFD76E",
                  },
                  "shadow": Object {
                    "blurry": "rgba(37,39,45,0.3)",
                    "border": "rgba(37,39,45,0.47)",
                  },
                  "text": Object {
                    "dark": "#0F0F10",
                    "default": "#25272D",
                    "disabled": "rgba(120,121,125,0.7)",
                    "muted": "#5B5D61",
                    "placeholder": "#989A9C",
                  },
                  "white": "#FFFFFF",
                },
                "fontSizes": Object {
                  "input": Object {
                    "default": "0.875rem",
                    "mobile": "1rem",
                  },
                  "size1": "0.6875rem",
                  "size2": "0.75rem",
                  "size3": "0.8125rem",
                  "size4": "0.875rem",
                  "size5": "1rem",
                  "size6": "1.25rem",
                  "size7": "1.5rem",
                  "size8": "1.75rem",
                  "size9": "2.125rem",
                },
                "fontWeights": Object {
                  "bold": 700,
                  "medium": 500,
                  "regular": 400,
                },
                "fonts": Object {
                  "default": "Motiva Sans",
                  "icon": "Arbor",
                },
                "iconFontPrefix": "ar",
                "icons": Object {
                  "annotation": "039",
                  "calendar": "035",
                  "caret": "037",
                  "chatBubbleOutline": "003",
                  "checkboxChecked": "030",
                  "checkboxIndeterminate": "02f",
                  "checkboxMarkedCircleOutline": "000",
                  "chevron": "03a",
                  "crLogo": "02e",
                  "cross": "03b",
                  "download": "004",
                  "swap": "038",
                  "symbolCircle": "034",
                  "thumbDownOutline": "001",
                  "thumbUpOutline": "002",
                  "trash": "036",
                },
                "lineHeights": Object {
                  "large": 1.5,
                  "regular": 1,
                  "small": 1.25,
                },
                "radii": Object {
                  "large": "6px",
                  "larger": "12px",
                  "small": "3px",
                },
                "shadows": Object {
                  "elevation0": "0 0 1px rgba(37,39,45,0.47)",
                  "elevation1": "0 0 1px rgba(37,39,45,0.47), 0 2px 4px -2px rgba(37,39,45,0.3)",
                  "elevation2": "0 0 1px rgba(37,39,45,0.47), 0 5px 8px -4px rgba(37,39,45,0.3)",
                  "elevation3": "0 0 1px rgba(37,39,45,0.47), 0 8px 10px -4px rgba(37,39,45,0.3)",
                  "elevation4": "0 0 1px rgba(37,39,45,0.47), 0 16px 24px -8px rgba(37,39,45,0.3)",
                },
                "space": Object {
                  "large": "1.5rem",
                  "larger": "2rem",
                  "largest": "4rem",
                  "regular": "1rem",
                  "small": "0.75rem",
                  "smaller": "0.5rem",
                  "smallest": "0.25rem",
                },
              }
            }
          >
            <EmotionClassNames>
              <StyledIcon
                className="ar ar-calendar"
                color="text.default"
                flex="0"
                fontSize="size4"
                ml="smaller"
                rotation={null}
              >
                <i
                  className="ar ar-calendar emotion-5 emotion-6"
                  color="text.default"
                  fontSize="size4"
                />
              </StyledIcon>
            </EmotionClassNames>
          </Icon>
        </WithTheme(Icon)>
      </ForwardRef>
    </div>
  </ProgressBarContainer>
</ProgressBar>
`;
